docs/latest/tutorial/tutorial-scaffolding.md

```diff
@@ -0,0 +1,354 @@
+    />
+    <meta
+      http-equiv="X-Content-Security-Policy"
+      content="default-src 'self'; script-src 'self'"
+    />
+    <title>Hello from Electron renderer!</title>
+  </head>
+  <body>
+    <h1>Hello from Electron renderer!</h1>
+    <p>👋</p>
+  </body>
+</html>
+```
+
+Now that you have a web page, load it into an application window. To do so, paste
+the following code in your `main.js` file so we can disect it line by line:
+
+```js title='main.js'
+const { app, BrowserWindow } = require('electron')
+
+const createWindow = () => {
+  const win = new BrowserWindow({
+    width: 800,
+    height: 600,
+    webPreferences: {
+      sandbox: true,
+    }
+  })
+
+  win.loadFile('index.html')
+}
+
+app.whenReady().then(() => {
+  createWindow()
+})
+```
+
+In the first line we are `require`ing 2 Electron modules:
+
+- The [`app`][app] module, which controls your application's event lifecycle.
+- The [`BrowserWindow`][browser-window] module, which creates and manages application
+  windows.
+
+```js
+const { app, BrowserWindow } = require('electron')
+```
+
+:::warning
+ESM (i.e.: use `import` to load a module) is currently not supported in Electron directly.
+You can find more information about the state and chanllenges in
+[this issue](https://github.com/electron/electron/issues/21457).
+:::
+
+Then, `createWindow()` function loads `index.html` into a new `BrowserWindow`
+instance with a series of options:
+
+```js
+const createWindow = () => {
+  const win = new BrowserWindow({
+    width: 800,
+    height: 600,
+    webPreferences: {
+      sandbox: true
+    }
+  })
+
+  win.loadFile('index.html')
+}
+```
+
+:::tip
+You can find all the options available in the [`BrowserWindow` page][browser-window].
+:::
+
+In Electron, browser windows can only be created after the `app` module's
+[`ready`][app-ready] event is fired. You can wait for this event by using the
+[`app.whenReady()`][app-when-ready] API. Call `createWindow()` after `whenReady()`
+resolves its Promise. This is done in the last 3 lines of the code:
+
+```js
+app.whenReady().then(() => {
+  createWindow()
+})
+```
+
+At this point, your Electron application should successfully open a window that displays your
+web page!
+
+You have now code being executed in the main and renderer process.
+The easiest way to debug the code in your renderer process is by using the built-in developer
+tools. You can invoke them with a shortcut (`Ctrl/Command+Shift+I`) or programmatically
+via `mainWindow.webContents.openDevTools()` in the main process. If you want to use VS Code,
+keep reading!
+
+The renderer has access to the web APIs. That means you can use the same JavaScript APIs and
+tooling you use for typical front-end development, such as using [`webpack`][webpack] to bundle
+and minify your code or [React][react] to manage your user interfaces.
+
+:::tip
+You can find more information about this subject in the [Boilerplates and CLIs].
+:::
+
+### Debugging from VS Code
```

Comment:
Is this meant to be an H3? I feel it'd fit better as an H2.

Reply:
```suggestion
### Optional: Debugging from VS Code
```

Reply:
```suggestion
## Optional: Debugging from VS Code
```